### DISCUSSION TITLE
Github Actions workflow file with typo

### DISCUSSION THREAD

**Discussion by seb-wahl** [community/community#26086] [Actions]
<p>Hi all<br>
I’m fairly new to Github Actions.<br>
I committed a workflow file (in .github/workflows) which had a typo. Under the Actions tab I’m told that the file has an error at line 62 and can’t be executed. However if I fix the file and commit it again the workflow is not run again and I also cannot run it manually.<br>
Any idea how to proceed. The repo can be found at <a href="https://github.com/esm-tools/esm_tools/actions?query=workflow%3A.github%2Fworkflows%2FCI-blogin-install-foci-default.yml" rel="nofollow noopener">https://github.com/esm-tools/esm_tools/actions?query=workflow%3A.github%2Fworkflows%2FCI-blogin-install-foci-default.yml</a><br>
Any idea how to “fix” this workflow? If I try to do a “delete workflow run”, I’m directed to a http 500 error page.</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/seb-wahl">@seb-wahl</a>,</p>
<p>I check your workflow file and find few syntax errors below:</p>
<ol>
<li>
<p>I see you set two ‘<strong>push</strong>’ event with different configurations in the same workflow.<br>
When using the ‘<strong>on</strong>’ key to set triggers, an event only can be used once in the same workflow, you can’t use the same event multiple times.<br>
If you want to a same event can trigger workflows with different configurations, you need to setup a workflow for each configuration, instead of using only one workflow.</p>
</li>
<li>
<p>The key ‘<strong>on</strong>’ is a keyword of the workflow syntax, you should not surround it with any other symbols (such as quotes or brackets). It is followed by the symbol colon (<strong>:</strong>).</p>
</li>
<li>
<p>When you use the ‘<strong>run</strong>’ key to set a step to run some command lines,</p>
<ul>
<li>
<p>if it only is a single line command, you can directly place the command and the ‘<strong>run</strong>’ key at the same line. For example:</p>
<pre><code class="lang-yaml">run: echo "There is only a single line."
</code></pre>
</li>
<li>
<p>if it is multiple command lines, you need to use the symbol ‘<strong>|</strong>’ to declare the multiple command lines. And the first command line also should be at its own line. For example:</p>
<pre><code class="lang-yaml">run: |
  echo "There is command line 1."
  echo "There is command line 2."
  . . .
  echo "There is command line N."
</code></pre>
<p>In addition, you do not need to use the quotes to surround the whole command lines.</p>
</li>
</ul>
</li>
</ol>
<p>More details about workflow syntax, see “<a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions" rel="nofollow noopener"><strong>Workflow syntax for GitHub Actions</strong></a>”.</p>
<p>Please try to fix these syntax errors to see if the problem still exists.<br>
The following is the workflow file I forked from yours, and I have fixed the syntax errors I can find. You can refer to it.<br>
<aside class="onebox githubblob">
  <header class="source">
      <a href="https://github.com/ForksForTest/esm_tools/blob/90c5748352b1007acb0945e8aa11b706247d4957/.github/workflows/CI-blogin-install-run-foci.yml" target="_blank" rel="nofollow noopener">github.com</a>
  </header>
  <article class="onebox-body">
    <h4><a href="https://github.com/ForksForTest/esm_tools/blob/90c5748352b1007acb0945e8aa11b706247d4957/.github/workflows/CI-blogin-install-run-foci.yml" target="_blank" rel="nofollow noopener">ForksForTest/esm_tools/blob/90c5748352b1007acb0945e8aa11b706247d4957/.github/workflows/CI-blogin-install-run-foci.yml</a></h4>
<pre><code class="lang-yml">name: CI-blogin-install-foci-default
on:
  push:
    branches:
    - refac_master_foci
  pull_request:
    branches:
    - release
    - develop
    - '*prep-release*'
    paths:
    - configs/setups/foci.yaml
    - configs/components/xios.yaml
    - configs/components/echam.yaml
    - configs/components/nemo.yaml
    - configs/components/oasis3mct/oasis3mct.yaml
    - configs/couplings/nemo-ORCA05_LIM2_KCM_AOW+echam-6.3.05p2-foci.yaml

jobs:
  install_models:
</code></pre>

  This file has been truncated. <a href="https://github.com/ForksForTest/esm_tools/blob/90c5748352b1007acb0945e8aa11b706247d4957/.github/workflows/CI-blogin-install-run-foci.yml" target="_blank" rel="nofollow noopener">show original</a>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
</p>

**Comment by seb-wahl** [comment]
<p>Thank you for your support. Yes once the syntax errors are removed, the actions are triggered again, and can also be run manually again. Thank you for putting me on the right track.</p>
